integrations-service/integrations/utils/integrations/mailgun.py

```diff
@@ -0,0 +1,63 @@
+import aiohttp
+from beartype import beartype
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ...autogen.Tools import MailgunSendEmailArguments, MailgunSetup
+from ...env import mailgun_api_key  # Import env to access environment variables
+from ...models import MailgunSendEmailOutput
+
+
+@beartype
+@retry(
+    wait=wait_exponential(multiplier=1, min=4, max=10),
+    reraise=True,
+    stop=stop_after_attempt(3),
+)
+async def send_email(
+    setup: MailgunSetup, arguments: MailgunSendEmailArguments
+) -> MailgunSendEmailOutput:
+    """
+    Sends an email with the provided details using the Mailgun API.
+    """
+
+    try:
+        domain = arguments.from_.split("@")[-1]
+    except Exception as e:
+        msg = f"Error splitting email domain: {e}"
+        raise Exception(msg)
+
+    # Extract domain from from_ address (e.g. "test@example.com" -> "email.example.com")
+    if not domain.startswith("email."):
+        domain = "email." + domain
+
+    # Use API key from env or from setup
+    api_key = mailgun_api_key if setup.api_key == "DEMO_API_KEY" else setup.api_key
```

Comment:
will there be any other conditions when the `api_key` will be assigned to the `mailgun_api_key`? For example, what if `setup.api_key` is empty?

Reply:
No. This check was typically added for demos.
When `api_key` is empty, it'll be sent as an empty string to mailgun, which in turn will not authorize the request, which is a normal behaviour.
When it's not set, a validation error will be thrown.